modules/standard/List.chpl

```diff
@@ -144,17 +180,121 @@ record list {
       delete current;
       current = next;
     }
+    first = nil;
+    last = nil;
+    length = 0;
   }
 
   pragma "no doc"
   proc writeThis(f) {
-    var first: bool = true;
+    var binary = f.binary();
+    var arrayStyle = f.styleElement(QIO_STYLE_ELEMENT_ARRAY);
+    var isspace = arrayStyle == QIO_ARRAY_FORMAT_SPACE && !binary;
+    var isjson = arrayStyle == QIO_ARRAY_FORMAT_JSON && !binary;
+    var ischpl = arrayStyle == QIO_ARRAY_FORMAT_CHPL && !binary;
+
+    if binary {
+      // Write the number of elements.
+      f <~> length;
+    }
+    if isjson || ischpl {
+      f <~> new ioLiteral("[");
+    }
+
+    var first = true;
     for e in this {
-      if !first then
-        f.write(" ");
-      else
-        first = false;
-      f.write(e);
+      if first then first = false;
+      else {
+        if isspace then f <~> new ioLiteral(" ");
+        else if isjson || ischpl then f <~> new ioLiteral(",");
```

Comment:
Should there be a space after the comma?  I think this would create [a,b,c] when we might prefer [a, b, c]